jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Callback.java

```diff
@@ -533,7 +572,7 @@ public void succeeded()
                 @Override
                 public void failed(Throwable x)
                 {
-                    callback.failed(x);
+                    Callback.failed(callback, x);
                     super.failed(x);
```

Comment:
Change this to have `super.failed()` inside a `finally`, so it gets called in all cases.